drivers/sensor/ti/ina700/ina700.c

```diff
@@ -0,0 +1,624 @@
+			return ret;
+		}
+	} else if (ina_attr == SENSOR_ATTR_CURRENT_OVER_LIMIT_THRESHOLD) {
+		ret = ina700_reg_read_16(&config->bus, INA700_REG_COL, &data);
+		if (ret < 0) {
+			return ret;
+		}
+	} else if (ina_attr == SENSOR_ATTR_CURRENT_UNDER_LIMIT_THRESHOLD) {
+		ret = ina700_reg_read_16(&config->bus, INA700_REG_CUL, &data);
+		if (ret < 0) {
+			return ret;
+		}
+	} else if (ina_attr == SENSOR_ATTR_BUS_OVERVOLTAGE_THRESHOLD) {
+		ret = ina700_reg_read_16(&config->bus, INA700_REG_BOVL, &data);
+		if (ret < 0) {
+			return ret;
+		}
+	} else if (ina_attr == SENSOR_ATTR_BUS_UNDERVOLTAGE_THRESHOLD) {
+		ret = ina700_reg_read_16(&config->bus, INA700_REG_BUVL, &data);
+		if (ret < 0) {
+			return ret;
+		}
+	} else if (ina_attr == SENSOR_ATTR_TEMPERATURE_OVER_LIMIT_THRESHOLD) {
+		ret = ina700_reg_read_16(&config->bus, INA700_REG_TEMP_LIMIT, &data);
+		if (ret < 0) {
+			return ret;
+		}
+	} else if (ina_attr == SENSOR_ATTR_POWER_OVER_LIMIT_THRESHOLD) {
+		ret = ina700_reg_read_16(&config->bus, INA700_REG_PWR_LIMIT, &data);
+		if (ret < 0) {
+			return ret;
+		}
+	} else {
+		LOG_ERR("INA700 attribute not supported.");
+		return -ENOTSUP;
+	}
+
+	val->val1 = data;
+	val->val2 = 0;
+
+	return 0;
+}
+
+/**
+ * @brief The work handler for the IN700 sensor trigger
+ * @param work a work item to retrieve the trigger from
+ * @return 0 if successful, negative errno code if failure.
+ */
+static void ina700_trigger_work_handler(struct k_work *work)
+{
+	struct ina700_trigger *trigg = CONTAINER_OF(work, struct ina700_trigger, conversion_work);
+	struct ina700_data *data = CONTAINER_OF(trigg, struct ina700_data, trigger);
+	const struct ina700_config *config = data->dev->config;
+	int ret;
+	uint16_t reg_alert;
+
+	/* Read reg alert to clear alerts */
+	ret = ina700_reg_read_16(&config->bus, INA700_REG_ALERT_DIAG, &reg_alert);
+	if (ret < 0) {
+		LOG_ERR("Failed to read alert register!");
+		return;
+	}
+
+	ret = ina700_read_data(data->dev);
+	if (ret < 0) {
+		LOG_WRN("Unable to read data, ret %d", ret);
+	}
+
+	if (data->trigger.handler_alert) {
+		data->trigger.handler_alert(data->dev, data->trigger.trig_alert);
+	}
+}
+
+/**
+ * @brief sensor operation mode check
+ *
+ * @retval true if set or false if not set
+ */
+static bool ina700_is_triggered_mode_set(const struct device *dev)
+{
+	const struct ina700_config *config = dev->config;
+
+	uint8_t mode = (config->adc_config & GENMASK(15, 12)) >> 12;
+
+	switch (mode) {
+	case INA700_OPER_MODE_TRIGGERED_BUS_VOLTAGE:
+	case INA700_OPER_MODE_TRIGGERED_TEMPERATURE:
+	case INA700_OPER_MODE_TRIGGERED_TEMPERATURE_BUS_VOLTAGE:
+	case INA700_OPER_MODE_TRIGGERED_TEMPERATURE_CURRENT:
+	case INA700_OPER_MODE_TRIGGERED_TEMPERATURE_CURRENT_BUS_VOLTAGE:
+		return true;
+	default:
+		return false;
+	}
+}
+
+/**
+ * @brief initializes the INA700 sensor
+ * @param dev the device to inititalize
+ * @return 0 if successful, negative errno code if failure.
+ */
+static int ina700_init(const struct device *dev)
```

Comment:
Generally on these init functions having all these `if (ret < 0)` adds a lot of runtime branches and in all cases the device is just not usable if they fail. They also make it impossible to unit test the init function. May I recommend a pattern we've used on Chrome OS:

```
ret = some_function();
__ASSERT(ret == 0, "The function failed");

ret |= some_other_function();
__ASSERT(ret == 0, "The other function failed");

if (ret != 0) {
  return -ENODEV;
}
return 0;
```